File: mac/NewtonMac/Models/Workspace.swift
//
//  Workspace.swift
//  Newton
//
//  Project & Workspace context model for isolated project folders and context.
//

import Foundation

public struct Workspace: Identifiable, Codable, Equatable, Hashable {
    public let id: String
    public var name: String
    public var iconName: String
    public var colorHex: String
    public var customSystemPrompt: String
    public var attachedFiles: [FileAttachment]
    public let createdAt: Date
    
    public init(
        id: String = UUID().uuidString,
        name: String,
        iconName: String = "folder.fill",
        colorHex: String = "#F5A623",
        customSystemPrompt: String = "",
        attachedFiles: [FileAttachment] = [],
        createdAt: Date = Date()
    ) {
        self.id = id
        self.name = name
        self.iconName = iconName
        self.colorHex = colorHex
        self.customSystemPrompt = customSystemPrompt
        self.attachedFiles = attachedFiles
        self.createdAt = createdAt
    }
}

File: android/shared/src/main/kotlin/ai/newton/shared/Workspace.kt
package ai.newton.shared

import kotlinx.serialization.Serializable
import java.util.UUID

/** Mirrors Swift `Workspace`. */
@Serializable
data class Workspace(
    val id: String = UUID.randomUUID().toString(),
    val name: String,
    val iconName: String = "folder.fill",
    val colorHex: String = "#F5A623",
    val customSystemPrompt: String = "",
    val attachedFiles: List<FileAttachment> = emptyList(),
    val createdAt: Long = System.currentTimeMillis(),
)

@Serializable
data class MemoryItem(
    val id: String = UUID.randomUUID().toString(),
    val content: String,
    val category: String = "general",
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis(),
)
